modules/geo-layers/src/trips-layer/trips-layer-uniforms.ts

```diff
@@ -0,0 +1,27 @@
+import {ShaderModule} from '@luma.gl/shadertools';
+import {UniformTypes} from '@deck.gl/core';
+
+const uniformBlock = `\
+uniform tripsUniforms {
+  bool fadeTrail;
+  float trailLength;
+  float currentTime;
+} trips;
+`;
+
+export type TripsProps = {
+  fadeTrail: boolean;
+  trailLength: number;
+  currentTime: number;
+};
+
+export const tripsUniforms = {
+  name: 'trips',
+  vs: uniformBlock,
+  fs: uniformBlock,
+  uniformTypes: {
+    fadeTrail: 'f32',
+    trailLength: 'f32',
+    currentTime: 'f32'
```

Comment:
Question, not blocking — Does the JS-side runtime code have any way to know that `fadeTrail` is a boolean? Or is that only known by the type-checker and GLSL?

Reply:
There should be a more elaborate optional prop type system in the shader module definition - it was used for run-time checks before we migrated to typescript. However I don't recommend adding that now without additional discussions. 

As for shader types for booleans I would recommend an integer rather than a float for obvious rounding reasons. We no longer have GLSL1 restrictions to worry about so no reason to prefer float over integer.